x-pack/plugins/security/server/saved_objects/index.ts

```diff
@@ -4,60 +4,47 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { IClusterClient, KibanaRequest, LegacyRequest } from '../../../../../src/core/server';
+import {
+  CoreSetup,
+  KibanaRequest,
+  LegacyRequest,
+  SavedObjectsClient,
+} from '../../../../../src/core/server';
 import { SecureSavedObjectsClientWrapper } from './secure_saved_objects_client_wrapper';
-import { LegacyAPI } from '../plugin';
 import { Authorization } from '../authorization';
 import { SecurityAuditLogger } from '../audit';
 
 interface SetupSavedObjectsParams {
-  adminClusterClient: IClusterClient;
   auditLogger: SecurityAuditLogger;
   authz: Pick<Authorization, 'mode' | 'actions' | 'checkSavedObjectsPrivilegesWithRequest'>;
-  legacyAPI: Pick<LegacyAPI, 'savedObjects'>;
+  savedObjects: CoreSetup['savedObjects'];
 }
 
-export function setupSavedObjects({
-  adminClusterClient,
-  auditLogger,
-  authz,
-  legacyAPI: { savedObjects },
-}: SetupSavedObjectsParams) {
+export function setupSavedObjects({ auditLogger, authz, savedObjects }: SetupSavedObjectsParams) {
   const getKibanaRequest = (request: KibanaRequest | LegacyRequest) =>
```

Comment:
__note:__ according to Rudolf we still need to keep this conversion code her as core doesn't do that for us even though typings for `setClientFactory` and `addClientWrapper` say opposite. It should be fixed soon and we can git rid of this function as well.